website/src/components/Tooltip/index.tsx

```diff
@@ -0,0 +1,49 @@
+import clsx from "clsx";
+import Link from "@docusaurus/Link";
+import styles from "./styles.module.css";
+import React, { useState, useEffect, ReactNode } from "react";
+import { keywordList, GlossaryTerm } from "./keywordList";
+
+interface TooltipProps {
+  className?: string;
+  children: ReactNode;
+}
+
+const Tooltip: React.FC<TooltipProps> = ({ className, children }) => {
+  const term = children?.toString();
+  const normalizedTerm = term?.toLowerCase().replace(/\s+/g, "-");
+  const [tooltipContent, setTooltipContent] = useState<string>("");
+  // Default link points to the Glossary
+  const [tooltipLink, setTooltipLink] = useState<string>(
+    `/docs/tech-explanation/glossary#${normalizedTerm}`
+  );
+
+  useEffect(() => {
```

Comment:
a little bit concern about `useEffect` here. can you check the academy implementation and see if it has better way to do this for performance? ref: https://react.dev/learn/you-might-not-need-an-effect